### PR TITLE
restore from a localized repo directly

### PR DESCRIPTION
**Pull request by OsmondJiang**
1. only implement the restore logic and a little build logic
2. add test cases against `restore`
3. need to discuss whether need support this behavior for `folder` mapping
4. add a lot of todos
5. will add more build logic like `bilingual` and `folder` mapping in another PR

### CODE REVIEW

File: docs/specs/localization.yml [hunk 1]
```diff
@@ -52,6 +52,74 @@ repos:
 outputs:
   docs/dep.json:
 ---
+# restore source repo when build from loc repo directly 1
```
Comment (live1206) [inline docs/specs/localization.yml]: Add description to avoid 1,2,3?

File: src/docfx/build/Build.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Repo
         {
             var errors = new List<Error>();
 
-            var (configErrors, config) = Config.Load(docsetPath, options);
+            var (configErrors, config) = LocalizationConvention.GetBuildConfig(docsetPath, options);
```
Comment (live1206) [inline src/docfx/build/Build.cs]: While building `en-us` repo, we also need to get the build config from `LocalizationConvention`?

File: docs/specs/localization.yml (hunk 1 above)
Comment (OsmondJiang) [reply docs/specs/localization.yml]: description is in second line, the first line is helping me to locate the failed test case quickly

Comment (live1206) [reply docs/specs/localization.yml]: ok

File: src/docfx/localization/LocalizationMapping.cs [hunk 3]
```diff
@@ -21,6 +21,7 @@ public enum LocalizationMapping
         /// /readme.md          -->         /localization/zh-cn/readme.md
         /// /files/a.md         -->         /localization/de-de/files/a.md
         /// </summary>
+        /// TODO: support build from localization folder directly?
```
Comment (live1206) [inline src/docfx/localization/LocalizationMapping.cs]: can be removed?

File: src/docfx/localization/LocalizationMapping.cs [hunk 4]
```diff
@@ -35,6 +36,9 @@ public enum LocalizationMapping
         /// #live                 -->           #live.zh-cn
         /// #live       -> bilingual ->         #live-sxs.zh-cn
         /// </summary>
+        /// TODO:
```
Comment (live1206) [inline src/docfx/localization/LocalizationMapping.cs]: can be removed?

File: src/docfx/localization/LocalizationMapping.cs (hunk 3 above)
Comment (OsmondJiang) [reply src/docfx/localization/LocalizationMapping.cs]: need to be discussed

File: src/docfx/localization/LocalizationMapping.cs (hunk 4 above)
Comment (OsmondJiang) [reply src/docfx/localization/LocalizationMapping.cs]: yes, will removed

**Pull request merged**